src/main/java/core/basesyntax/FileWork.java

```diff
@@ -8,7 +17,28 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
-    public String[] readFromFile(String fileName) {
-        return null;
+
+    private static final String SEARCH_WORD = "\\b[w]\\w*\\b";
+
+    public static String[] readFromFile(String fileName) {
+        List<String> wordList = new ArrayList<>();
+        Pattern pattern = Pattern.compile(SEARCH_WORD);
+        Matcher matcher;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                matcher = pattern.matcher(line.toLowerCase());
+                while (matcher.find()) {
+                    wordList.add(matcher.group());
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("File not found!");
```

Comment:
let's also pass exception object to constructor(overloaded) in order not to lose information about what has happened